src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -0,0 +1,468 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// Contains checks that are not architecture-specific
+
+#![allow(dead_code)]
+
+use crate::arch::arch_specific;
+use crate::args::EnvArgument;
+use crate::check;
+use crate::ops::version;
+use crate::utils;
+use kata_types::config::TomlConfig;
+
+use anyhow::{anyhow, Context, Result};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+use sys_info;
+
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct HostInfo {
+    #[serde(default)]
+    available_guest_protection: String,
+    #[serde(default)]
+    kernel: String,
+    #[serde(default)]
+    architecture: String,
+    #[serde(default)]
+    vm_container_capable: bool,
+    #[serde(default)]
+    support_v_socks: bool,
```

Comment:
Nit: I'd consider renaming to:

```suggestion
    support_vsocks: bool,
```

Reply:
fixed